crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs

```diff
@@ -214,18 +218,46 @@ fn scalar_value_to_datum(value: &ScalarValue) -> Option<Datum> {
         ScalarValue::LargeUtf8(Some(v)) => Some(Datum::string(v.clone())),
         ScalarValue::Date32(Some(v)) => Some(Datum::date(*v)),
         ScalarValue::Date64(Some(v)) => Some(Datum::date((*v / MILLIS_PER_DAY) as i32)),
+        ScalarValue::TimestampSecond(Some(v), tz) => {
+            interpret_timestamptz_micros(v.checked_mul(MICROS_PER_SECOND)?, tz.as_deref())
+        }
+        ScalarValue::TimestampMillisecond(Some(v), tz) => {
+            interpret_timestamptz_micros(v.checked_mul(MICROS_PER_MILLISECOND)?, tz.as_deref())
+        }
+        ScalarValue::TimestampMicrosecond(Some(v), tz) => {
+            interpret_timestamptz_micros(*v, tz.as_deref())
+        }
+        ScalarValue::TimestampNanosecond(Some(v), Some(_)) => Some(Datum::timestamptz_nanos(*v)),
+        ScalarValue::TimestampNanosecond(Some(v), None) => Some(Datum::timestamp_nanos(*v)),
         _ => None,
     }
 }
 
+fn interpret_timestamptz_micros(micros: i64, tz: Option<impl AsRef<str>>) -> Option<Datum> {
+    let offset = tz
+        .as_ref()
+        .and_then(|s| s.as_ref().parse::<FixedOffset>().ok());
+
+    match offset {
+        Some(off) => off
+            .timestamp_micros(micros)
+            .single()
+            .map(Datum::timestamptz_from_datetime),
```

Comment:
double checking my understanding here, does this normalize micros with offset to UTC? 

Iceberg Datums are in UTC in `timestamptz_from_datetime` 
https://github.com/apache/iceberg-rust/blob/0107f17f568c770d22c13aab9f38f1726df38745/crates/iceberg/src/spec/values/datum.rs#L890-L902